From 04ccc9856138825a56e48a35ecfa5298c7877af6 Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Wed, 26 Jun 2019 17:44:13 +0200
Subject: [PATCH 3/8] Revert "futex: Ensure lock/unlock symetry versus pi_lock
 and hash bucket lock"

commit 6a773b70cf105b46298ed3b44e77c102ce31d9ec from linux-rt-devel

Drop the RT fixup, the futex code will be changed to avoid the need for
the workaround.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
[OP: backport of 6a773b70cf10 from linux-5.0.y-rt branch of linux-rt-devel]
Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 kernel/futex.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/futex.c b/kernel/futex.c
index ca6af8e50ef0..0a01f0301bc3 100644
--- a/kernel/futex.c
+++ b/kernel/futex.c
@@ -918,9 +918,7 @@ void exit_pi_state_list(struct task_struct *curr)
 		if (head->next != next) {
 			/* retain curr->pi_lock for the loop invariant */
 			raw_spin_unlock(&pi_state->pi_mutex.wait_lock);
-			raw_spin_unlock_irq(&curr->pi_lock);
 			spin_unlock(&hb->lock);
-			raw_spin_lock_irq(&curr->pi_lock);
 			put_pi_state(pi_state);
 			continue;
 		}
-- 
2.17.1

